build.sh

```
@@ -199,6 +199,30 @@ build_cpp() {
+
+    if [ ! -d "$BUILD_DIR" ]; then
+        echo "Build directory not found, running cmake first..."
+        mkdir -p "$BUILD_DIR"
+        cd "$BUILD_DIR"
+        cmake .. -DCMAKE_BUILD_TYPE=Release
+    else
+        cd "$BUILD_DIR"
+    fi
+
+    echo "Building rag_analyzer_c_test..."
+    make rag_analyzer_c_test -j$(nproc)
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)